src/tests/ftest/server/daos_server_restart.py

```diff
@@ -0,0 +1,187 @@
+            uuid = result['uuid']
+            svc = result['svc']
+            for _ in range(container_per_pool):
+                daos_cmd = DaosCommand(self.bin)
+                daos_cmd.exit_status_exception = False
+                result = daos_cmd.container_create(pool=uuid, svc=svc)
+                self.log.info("container create status: %s", result)
+
+    def test_daos_server_reformat(self):
+        """JIRA ID: DAOS-3596.
+
+        Test Description: verify reformatting DAOS is same as a fresh install.
+        Steps:
+        (1)Verify daos server pool list after started.
+        (2)Restart server without pool been created, and verify.
+        (3)Create pools, containers and verify the pools.
+        (4)Perform an orderly DAOS shutdown, and Use cmd line tools to
+           re-format both SSDs and NVDIMMs.
+        (5)Verify after DAOS server restarted, it should appear as an empty
+           fresh installation.
+
+        :avocado: tags=all,pr,hw,large,server_test,server_reformat
+        """
+
+        self.log.info("(1)Verify daos server pool list after started.")
+        self.verify_pool_list()
+        self.log.info("(2)Restart server without pool created and verify.")
+        self.restart_daos_server()
+        self.verify_pool_list()
+        self.log.info("(3)Create pools, containers.")
+        self.create_pool_and_container()
+        self.log.info("(4)Shutdown, restart and reformat the server")
+        self.restart_daos_server()
+        self.log.info("(5)Verify after server restarted.")
+        self.verify_pool_list()
+
+    def test_daos_io_server_restart(self):
+        """
+        JIRA ID: DAOS-3593.
+
+        Test Description: Test cmd to perform daos io_server restart.
+        Steps:
+        (1)Use the cmd line to perform a controlled shutdown from a
+           quiescent state (i.e. clients disconnected).
+        (2)Use the cmd line to perform a controlled shutdown from a
+           partially quiescent state (i.e. clients attached but no
+           transactions in progress).
+        (3)Force shutdown and restart the daos io-server.
+        (4)Verify pool list after forced shutdown and restart the
+           daos io-server.
+        (5)Use the cmd line to perform a controlled shutdown when the
+           daos cluster is incomplete (i.e. 1 of the 2 servers is down).
+
+        :avocado: tags=all,pr,hw,large,server_test,server_restart
+        """
+
+        self.log.info(
+            "(1)Shutdown and restart the daos io-server "
+            "from a quiescent state.")
+        self.agent_managers[0].stop()
+        self.verify_pool_list()
+        self.restart_daos_io_server()
+        self.agent_managers[0].start()
+        self.log.info(
+            "(2)Shutdown and restart the daos io-server with pools "
+            "and containers created.")
+        self.create_pool_and_container()
+        pool_list = self.get_pool_list()
+        self.restart_daos_io_server()
```

Comment:
unless I'm mistaken, system stop isn't expected to work reliably without `--force` currently

Reply:
We are testing both with and without --force restart daos io-server.
There is a bug reported (DAOS-5392 -> DAOS-3883) for the issue of segmentation fault when restart io-server without --force.

Reply:
leaving system stop without force is likely to introduce intermittent failures into CI , which is not desirable

Reply:
ok. Thanks for the comments.
I will comment out the test steps without the --force to avoid the intermittent CI failure.
Please see update on the next commit.